tests/integration/assessment/test_workflows.py

```diff
@@ -4,21 +4,46 @@
 from databricks.sdk.retries import retried
 from databricks.sdk.service.iam import PermissionLevel
 
+from databricks.labs.ucx.hive_metastore import TablesCrawler
+
 
 @retried(on=[NotFound, InvalidParameterValue], timeout=timedelta(minutes=8))
-def test_running_real_assessment_job(ws, installation_ctx, make_cluster_policy, make_cluster_policy_permissions):
+def test_running_real_assessment_job(
+    installation_ctx,
+    make_cluster_policy,
+    make_cluster_policy_permissions,
+    make_schema,
+    make_table,
+    sql_backend,
+    inventory_schema,
+):
     ws_group_a, _ = installation_ctx.make_ucx_group()
     cluster_policy = make_cluster_policy()
     make_cluster_policy_permissions(
         object_id=cluster_policy.policy_id,
         permission_level=PermissionLevel.CAN_USE,
         group_name=ws_group_a.display_name,
     )
+
+    source_schema = make_schema(catalog_name="hive_metastore")
+    managed_table = make_table(schema_name=source_schema.name)
+    external_table = make_table(schema_name=source_schema.name, external=True)
+    tmp_table = make_table(schema_name=source_schema.name, ctas="SELECT 2+2 AS four")
+    view = make_table(schema_name=source_schema.name, ctas="SELECT 2+2 AS four", view=True)
+    non_delta = make_table(schema_name=source_schema.name, non_delta=True)
+
     installation_ctx.__dict__['include_object_permissions'] = [f"cluster-policies:{cluster_policy.policy_id}"]
     installation_ctx.workspace_installation.run()
 
     installation_ctx.deployed_workflows.run_workflow("assessment")
-    assert installation_ctx.deployed_workflows.validate_step("assessment")
 
     after = installation_ctx.generic_permissions_support.load_as_dict("cluster-policies", cluster_policy.policy_id)
     assert after[ws_group_a.display_name] == PermissionLevel.CAN_USE
+
+    tables = []
+    for _ in TablesCrawler(sql_backend, inventory_schema, [source_schema.name]).snapshot():
+        tables.append(_.name)
+
+    expected_tables = {managed_table.name, external_table.name, tmp_table.name, view.name, non_delta.name}
+    assert len(tables) == len(expected_tables)
+    assert set(tables) == expected_tables
```

Comment:
```suggestion
    tables = set[str]()
    local_crawler = TablesCrawler(sql_backend, inventory_schema, [source_schema.name])
    for _ in local_crawler.snapshot():
        tables.add(_.name)

    expected_tables = {managed_table.name, external_table.name, tmp_table.name, view.name, non_delta.name}
    assert tables == expected_tables
```

use correct datastructures